lld/COFF/LTO.cpp

```diff
@@ -213,6 +221,8 @@ std::vector<InputFile *> BitcodeCompiler::compile() {
     pruneCache(ctx.config.ltoCache, ctx.config.ltoCachePolicy, files);
 
   std::vector<InputFile *> ret;
+  bool emitASM = ctx.config.emit == EmitKind::ASM;
+  const char *Ext = emitASM ? ".asm" : ".obj";
```

Comment:
I believe we could hook this up to generate masm style assembly instead, would that be preferable?
The ELF linker even has the option to choose between styles, but if we went that direction, I lean on doing it on a separate MR.

Reply:
That's definitely a separate MR, yeah.

For x86, I guess it could be possible to generate masm style assembly - for ARM, the MS tooling uses "armasm" style assembly, and I don't think we've got support for either generating or consuming that anywhere.

So before that, I'd say we'd go with just `.s` as we're dealing with GAS style assembly.